src/lib/censorRecords.js

```diff
@@ -0,0 +1,208 @@
+const censor = (blocked, roles) => (redact, value) => {
+  return blocked !== false || (redact && !roles.includes('full-details')) ?
+    null : value
+}
+
+const censorDeath = (death, roles) => {
```

Comment:
I was thinking that it would actually be nicer to use another closure here:
```js
const censorDeath = roles => death => {
```
That way it would be nicer to map with.
But we can do that another time.

Reply:
Discussed and agreed to leave